src/librados/librados_c.cc

```diff
@@ -666,6 +668,135 @@ extern "C" int _rados_inconsistent_pg_list(rados_t cluster, int64_t pool_id,
 }
 LIBRADOS_C_API_BASE_DEFAULT(rados_inconsistent_pg_list);
 
+static void dump_inconsistent(const librados::inconsistent_obj_t& inc,
+            Formatter& f)
+{
+  f.dump_string("obj_name", inc.object.name);
+
+  f.open_array_section("errors");
+  if (inc.has_object_info_inconsistency())
+    f.dump_string("error", "object_info_inconsistency");
+  if (inc.has_data_digest_mismatch())
+    f.dump_string("error", "data_digest_mismatch");
+  if (inc.has_omap_digest_mismatch())
+    f.dump_string("error", "omap_digest_mismatch");
+  if (inc.has_size_mismatch())
+    f.dump_string("error", "size_mismatch");
+  if (inc.has_attr_value_mismatch())
+    f.dump_string("error", "attr_value_mismatch");
+  if (inc.has_attr_name_mismatch())
+    f.dump_string("error", "attr_name_mismatch");
+  if (inc.has_snapset_inconsistency())
+    f.dump_string("error", "snapset_inconsistency");
+  if (inc.has_hinfo_inconsistency())
+    f.dump_string("error", "hinfo_inconsistency");
+  f.close_section(); // errors
+
+  f.open_array_section("shards");
+  for (const auto& shard_info : inc.shards) {
+    f.open_object_section("shard");
+    auto& osd_shard = shard_info.first;
+    f.dump_int("osd", osd_shard.osd);
+    f.dump_bool("primary", shard_info.second.primary);
+    f.close_section(); // shard
+  }
+  f.close_section(); // shards
+}
+
+extern "C" int _rados_inconsistent_obj_list(rados_t cluster,
+            const char* pg_str, char* buf, size_t size)
+{
+  tracepoint(librados, rados_inconsistent_obj_list_enter, cluster, pg_str, size);
+  librados::PlacementGroup pg;
+  int ret = pg.parse(pg_str);
+  if (!ret) {
+    return -EINVAL;
+  }
+
+  librados::Rados rados;
+  ret = rados.init_with_context(g_ceph_context);
```

Comment:
this is very unusual. i don't think we should do this kind of setup/teardown (which is missing by the way) in a function like this.

Reply:
right, `rados_ioctx_t` should be passed instead of `rados_t`